ports/stm32/machine_i2c.c

```diff
@@ -44,28 +44,29 @@ typedef struct _machine_hard_i2c_obj_t {
     i2c_t *i2c;
     mp_hal_pin_obj_t scl;
     mp_hal_pin_obj_t sda;
+    uint32_t timeout;
```

Comment:
This won't work because the objects below are defined "const" (to not use RAM, they live in ROM).  Best way to deal with it is to have a separate non-const array to hold just the timeouts (the mutable data).

And then have this as a pointer to an entry in that array: `uint32_t *timeout`.